functions/wait/action/kubernetesUpgrade.go

```diff
@@ -5,108 +5,57 @@ import (
-				return false, err
-			}
-
-			if cluster.Version.GitVersion == expectedUpgradedKubernetesVersion {
-				t.Logf("Successfully updated kubernetes version to %v", clusterConfig.UpgradedKubernetesVersion)
-				return true, nil
-			}
-
-			if !waiting {
-				t.Logf("Waiting for cluster to upgrade kubernetes version...")
-				waiting = true
-			}
-
-			return false, nil
-		})
-		require.NoError(t, waitErr)
+const (
+	aks        = "aks"
+	eks        = "eks"
+	gke        = "gke"
+	ec2RKE1    = "ec2_rke1"
+	ec2RKE2    = "ec2_rke2"
+	ec2K3s     = "ec2_k3s"
+	linodeRKE1 = "linode_rke1"
+	linodeRKE2 = "linode_rke2"
+	linodeK3s  = "linode_k3s"
+)
 
-		if waitErr != nil {
-			t.Logf("Failed to instantiate kubernetes upgrade wait poll.")
-			return false, waitErr
-		}
+// KubernetesUpgrade is a function that will upgrade the Kubernetes version of the provisioned cluster.
+func KubernetesUpgrade(t *testing.T, client *rancher.Client, clusterID string, module string, clusterConfig *config.TerratestConfig) (done bool, err error) {
+	var expectedUpgradedKubernetesVersion string
+
+	switch {
+	case module == aks || module == eks || module == gke:
+		expectedUpgradedKubernetesVersion = `v` + clusterConfig.UpgradedKubernetesVersion
+	case module == ec2RKE1 || module == linodeRKE1:
+		expectedUpgradedKubernetesVersion = clusterConfig.UpgradedKubernetesVersion[:len(clusterConfig.UpgradedKubernetesVersion)-11]
+	case module == ec2RKE2 || module == ec2K3s || module == linodeRKE2 || module == linodeK3s:
+		expectedUpgradedKubernetesVersion = clusterConfig.UpgradedKubernetesVersion
+	default:
+		logrus.Errorf("Unsupported module: %v", module)
 	}
 
-	if module == "ec2_rke1" || module == "linode_rke1" {
-		expectedUpgradedKubernetesVersion := clusterConfig.UpgradedKubernetesVersion[:len(clusterConfig.UpgradedKubernetesVersion)-11]
-		waiting := false
-		waitErr := wait.Poll(100*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
-			cluster, clientErr := client.Management.Cluster.ByID(clusterID)
-			require.NoError(t, err)
-
-			if clientErr != nil {
-				t.Logf("Failed to locate cluster and grab client specs. Error: %v", err)
-				return false, err
-			}
-
-			if cluster.Version.GitVersion == expectedUpgradedKubernetesVersion {
-				return true, nil
-			}
+	waiting := false
+	err = wait.Poll(100*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
+		cluster, err := client.Management.Cluster.ByID(clusterID)
+		require.NoError(t, err)
 
-			if !waiting {
-				t.Logf("Waiting for cluster to upgrade kubernetes version...")
-				waiting = true
-			}
-
-			return false, nil
-		})
-		require.NoError(t, waitErr)
-
-		if waitErr != nil {
-			t.Logf("Failed to instantiate kubernetes upgrade wait poll.")
-			return false, waitErr
+		if cluster.Version.GitVersion == expectedUpgradedKubernetesVersion {
+			logrus.Infof("Successfully updated kubernetes version to %v!", clusterConfig.UpgradedKubernetesVersion)
+			return true, nil
 		}
```

Comment:
wait.Poll is being deprecated unless i'm greatly mistaken, since your touching these files could you update this?

Reply:
This is part of the go.mod issue that was discussed in our team meeting. Until that is bumped up, separate from this PR, we have to keep it as is. But this is on my radar.